// @atproto/api/dist/client/types/tools/ozone/report/reassignQueue.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueueDisabledError = exports.QueueNotFoundError = exports.AlreadyInTargetQueueError = exports.ReportClosedError = exports.ReportNotFoundError = void 0;
exports.toKnownErr = toKnownErr;
/**
 * GENERATED CODE - DO NOT MODIFY
 */
const xrpc_1 = require("@atproto/xrpc");
const lexicons_1 = require("../../../../lexicons");
const util_1 = require("../../../../util");
const is$typed = util_1.is$typed, validate = lexicons_1.validate;
const id = 'tools.ozone.report.reassignQueue';
class ReportNotFoundError extends xrpc_1.XRPCError {
    constructor(src) {
        super(src.status, src.error, src.message, src.headers, { cause: src });
    }
}
exports.ReportNotFoundError = ReportNotFoundError;
class ReportClosedError extends xrpc_1.XRPCError {
    constructor(src) {
        super(src.status, src.error, src.message, src.headers, { cause: src });
    }
}
exports.ReportClosedError = ReportClosedError;
class AlreadyInTargetQueueError extends xrpc_1.XRPCError {
    constructor(src) {
        super(src.status, src.error, src.message, src.headers, { cause: src });
    }
}
exports.AlreadyInTargetQueueError = AlreadyInTargetQueueError;
class QueueNotFoundError extends xrpc_1.XRPCError {
    constructor(src) {
        super(src.status, src.error, src.message, src.headers, { cause: src });
    }
}
exports.QueueNotFoundError = QueueNotFoundError;
class QueueDisabledError extends xrpc_1.XRPCError {
    constructor(src) {
        super(src.status, src.error, src.message, src.headers, { cause: src });
    }
}
exports.QueueDisabledError = QueueDisabledError;
function toKnownErr(e) {
    if (e instanceof xrpc_1.XRPCError) {
        if (e.error === 'ReportNotFound')
            return new ReportNotFoundError(e);
        if (e.error === 'ReportClosed')
            return new ReportClosedError(e);
        if (e.error === 'AlreadyInTargetQueue')
            return new AlreadyInTargetQueueError(e);
        if (e.error === 'QueueNotFound')
            return new QueueNotFoundError(e);
        if (e.error === 'QueueDisabled')
            return new QueueDisabledError(e);
    }
    return e;
}
