// @atproto/api/dist/client/types/tools/ozone/queue/defs.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isQueueView = isQueueView;
exports.validateQueueView = validateQueueView;
exports.isQueueStats = isQueueStats;
exports.validateQueueStats = validateQueueStats;
exports.isAssignmentView = isAssignmentView;
exports.validateAssignmentView = validateAssignmentView;
const lexicons_1 = require("../../../../lexicons");
const util_1 = require("../../../../util");
const is$typed = util_1.is$typed, validate = lexicons_1.validate;
const id = 'tools.ozone.queue.defs';
const hashQueueView = 'queueView';
function isQueueView(v) {
    return is$typed(v, id, hashQueueView);
}
function validateQueueView(v) {
    return validate(v, id, hashQueueView);
}
const hashQueueStats = 'queueStats';
function isQueueStats(v) {
    return is$typed(v, id, hashQueueStats);
}
function validateQueueStats(v) {
    return validate(v, id, hashQueueStats);
}
const hashAssignmentView = 'assignmentView';
function isAssignmentView(v) {
    return is$typed(v, id, hashAssignmentView);
}
function validateAssignmentView(v) {
    return validate(v, id, hashAssignmentView);
}
